### PR TITLE
update to rc.6 and use vitejs/plugin-react-swc

### PR DESCRIPTION
**Pull request by jwartofsky-yext**
prod and dev commands run successfully. Pages load as expected.

### CODE REVIEW

File: src/components/DirectoryCityGrid.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import * as React from "react";
-import { Address } from "@yext/pages/components";
+import { Address } from "@yext/sites-components";
```
Comment (asanehisa) [inline src/components/DirectoryCityGrid.tsx:2]: I think we did the rename to pages-components

Comment (jwartofsky-yext) [reply src/components/DirectoryCityGrid.tsx:2]: We did, but it hasn't released yet

**Comment by asanehisa** [comment]
Update to rc7 and vite 5 now that release is cut! site-components -> pages-components is also released.

File: package.json [hunk 2]
```diff
@@ -6,23 +6,17 @@
...
+    "build": "pages build"
   },
   "engines": {
-    "node": ">=17"
+    "node": "^18.0.0 || >=20.0.0"
   },
   "dependencies": {
     "@headlessui/react": "^1.7.12",
     "@heroicons/react": "^2.0.14",
+    "@yext/sites-components": "1.0.0-rc.5",
```
Comment (mkilpatrick) [inline package.json:19]: Should be pages-components. Add caret to everything. Update all other deps besides react. Use pagesjs rc.7.

File: package.json [hunk 3]
```diff
@@ -34,8 +28,8 @@
     "@types/react-dom": "^17.0.17",
     "@typescript-eslint/eslint-plugin": "^5.36.2",
     "@typescript-eslint/parser": "^5.36.2",
-    "@vitejs/plugin-react": "^2.1.0",
-    "@yext/pages": "1.0.0-beta.21",
+    "@yext/pages": "1.0.0-rc.6",
+    "@vitejs/plugin-react-swc": "^3.5.0",
```
Comment (mkilpatrick) [inline package.json:32]: Remove generate-license-file

**Comment by jwartofsky-yext** [comment]
moved to YextSolutions

**Pull request closed without merging**